### ISSUE TITLE
The ability to use ReactNode as label prop of FormTab

### ISSUE THREAD

**Issue by dejancg** [marmelab/react-admin#5436]
Currently, the **FormTab** component declares `label` prop as `string` type ([here](https://github.com/marmelab/react-admin/blob/801e2456844c11325504c7d496048c7c32b557be/packages/ra-ui-materialui/src/form/FormTab.tsx#L93)):
```ts
export interface FormTabProps {
    // ...
    label: string;
    // ..
}
```
However, the **FormTab** component merely passes the prop value to **MuiTab** component (translated though), as seen [here](https://github.com/marmelab/react-admin/blob/801e2456844c11325504c7d496048c7c32b557be/packages/ra-ui-materialui/src/form/FormTab.tsx#L36):
```ts
const renderHeader = () => (
    <MuiTab
        // ...
        label={translate(label, { _: label })}
        // ...
    />
);
```
And the **MuiTab** component declares the `label` as `ReactNode`, as seen [here](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Tab/Tab.d.ts#L54): 
```ts
export type TabTypeMap<P = {}, D extends React.ElementType = 'div'> = ExtendButtonBaseTypeMap<{
  props: P & {
    // ...
    /**
     * The label element.
     */
    label?: React.ReactNode;
    // ...
  };
  // ...
}>;
```

To work around this limitation, I did this:
1. Created a type guard to assert the label type:
```ts
const isString = (element: any): element is string => {
  return typeof element === "string" || element instanceof String;
};
```
2. Changed the `renderHeader` func like so:
```ts
  const renderHeader = () => (
    <MuiTab
      key={isString(label) ? label : label.props.key} // <-- Might be a better idea to use the FormTab key here
      label={isString(label) ? translate(label, { _: label }) : label}
     // ...
    />
  );
```
3. Change the FormTabProps `label` declaration like so:
```ts
export interface FormTabProps {
  // ...
  label: string | ReactElement;
  // ...
}
```
The obvious downside is that I lost the ability to have my translation key translated automatically when an element is used instead of plain string. But other than that, seems to be working fine. Any thoughts?

**Comment by djhi** [comment]
Thanks for reporting. Marking this as an enhancement

**Comment by djhi** [comment]
Fixed by #6061
